src/mesh/voxel_cloud.rs

```diff
@@ -995,6 +1015,45 @@ impl ScalarField {
         }
     }
 
+    /// Add values of the other scalar fields to the respective voxels of the
+    /// current scalar field. The current scalar field will not be resized, so
+    /// only the overlapping areas will be will be added. If the voxel in the
+    /// current or the other scalar field is None (does not contain any voxel or
+    /// does not exist), it is treated according to the selected
+    /// `empty_voxels_treatment` - it is either replaced with a constant value
+    /// or considered None and then also the resulting voxel will become None.
```

Comment:
This description should be on the `TreatEmpty` type and the documentation here should just say something like: "Empty voxels of both scalar fields are treated according to the selected `empty_voxels_treatment` parameter."

Also I think we will want different `TreatEmpty` for self and other in the future.

Reply:
maybe... but i can't imagine a situation when id' like to treat self and other differently. i prefer just leaving a FIXME note

Reply:
so if the falloff functions happen to contain the logic, shouldn't this be their param?

Reply:
Seems like independent parameters to me. Also I can't find where the falloff functions contain this logic. Regardless of whether the falloff functions are implemented in an impl block or inline, the value from the other voxel cloud is extracted beforehand.

Reply:
rubberducking:
```
treat_self = AsNone, treat_other = AsNone {
1+1 = 2
None + 1 = None
1 + None = None
}

treat_self = AsValue(1), treat_other = AsNone {
1+1 = 2
None + 1 = 2
1 + None = None
}

treat_self = AsNone, treat_other = AsValue(1) {
1+1 = 2
None + 1 = None
1 + None = 2
}

treat_self = AsValue(1), treat_other = AsValue(1) {
1+1 = 2
None + 1 = 2
1 + None = 2
}
```

hmm, okay

Reply:
Just now noticed this. I would personally treat `None` as zero for `AsNone`, so:

None + None = 0
1 + None = 1
None + 1 = 1